From 8376b689d423fa039824bb905b9749c3356809dc Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Tue, 11 Nov 2014 18:15:55 +0800
Subject: [PATCH 382/602] MA-5839-4 Update android to Lollipop

Unify the feature list for sabreasd imx6dq with below rules:
1. Reference the tablet_core_hardware.xml features if the board can support them
2. Include the feature file if it exist in frameworks/native/data/etc
3. Add those feature into required_hardware.xml if it cannot be
find in framework/native/data/etc

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 imx6/sabresd_6dq.mk               | 10 +++++-----
 sabresd_6dq/required_hardware.xml | 22 ----------------------
 2 files changed, 5 insertions(+), 27 deletions(-)

diff --git a/imx6/sabresd_6dq.mk b/imx6/sabresd_6dq.mk
index edde8a4..1087d6f 100644
--- a/imx6/sabresd_6dq.mk
+++ b/imx6/sabresd_6dq.mk
@@ -17,7 +17,6 @@ PRODUCT_NAME := sabresd_6dq
 PRODUCT_DEVICE := sabresd_6dq
 
 PRODUCT_COPY_FILES += \
-	device/fsl/sabresd_6dq/required_hardware.xml:system/etc/permissions/required_hardware.xml \
 	device/fsl/sabresd_6dq/init.rc:root/init.freescale.rc \
 	device/fsl/sabresd_6dq/gpsreset.sh:system/etc/gpsreset.sh \
 	device/fsl/sabresd_6dq/audio_policy.conf:system/etc/audio_policy.conf \
@@ -36,18 +35,19 @@ PRODUCT_CHARACTERISTICS := tablet
 PRODUCT_AAPT_CONFIG += xlarge large tvdpi hdpi
 
 PRODUCT_COPY_FILES += \
+	frameworks/native/data/etc/tablet_core_hardware.xml:system/etc/permissions/tablet_core_hardware.xml \
+	frameworks/native/data/etc/android.hardware.camera.xml:system/etc/permissions/android.hardware.camera.xml \
+	frameworks/native/data/etc/android.hardware.camera.front.xml:system/etc/permissions/android.hardware.camera.front.xml \
 	frameworks/native/data/etc/android.hardware.location.gps.xml:system/etc/permissions/android.hardware.location.gps.xml \
 	frameworks/native/data/etc/android.hardware.wifi.xml:system/etc/permissions/android.hardware.wifi.xml \
 	frameworks/native/data/etc/android.hardware.wifi.direct.xml:system/etc/permissions/android.hardware.wifi.direct.xml \
 	frameworks/native/data/etc/android.hardware.sensor.light.xml:system/etc/permissions/android.hardware.sensor.light.xml \
-	frameworks/native/data/etc/android.hardware.sensor.accelerometer.xml:system/etc/permissions/android.hardware.sensor.accelerometer.xml \
-	frameworks/native/data/etc/android.hardware.faketouch.xml:system/etc/permissions/android.hardware.faketouch.xml \
-	frameworks/native/data/etc/android.hardware.touchscreen.multitouch.jazzhand.xml:system/etc/permissions/android.hardware.touchscreen.multitouch.jazzhand.xml \
 	frameworks/native/data/etc/android.software.sip.voip.xml:system/etc/permissions/android.software.sip.voip.xml \
 	frameworks/native/data/etc/android.hardware.usb.host.xml:system/etc/permissions/android.hardware.usb.host.xml \
 	frameworks/native/data/etc/android.hardware.usb.accessory.xml:system/etc/permissions/android.hardware.usb.accessory.xml \
 	frameworks/native/data/etc/android.hardware.bluetooth_le.xml:system/etc/permissions/android.hardware.bluetooth_le.xml \
-	frameworks/native/data/etc/android.hardware.consumerir.xml:system/etc/permissions/android.hardware.consumerir.xml
+	frameworks/native/data/etc/android.hardware.consumerir.xml:system/etc/permissions/android.hardware.consumerir.xml \
+	device/fsl/sabresd_6dq/required_hardware.xml:system/etc/permissions/required_hardware.xml
 
 # for PDK build, include only when the dir exists
 # too early to use $(TARGET_BUILD_PDK)
diff --git a/sabresd_6dq/required_hardware.xml b/sabresd_6dq/required_hardware.xml
index 1c1dbc2..b624650 100644
--- a/sabresd_6dq/required_hardware.xml
+++ b/sabresd_6dq/required_hardware.xml
@@ -19,28 +19,6 @@
 -->
 <permissions>
     <feature name="android.hardware.camera.any" />
-    <feature name="android.hardware.camera" />
-    <feature name="android.hardware.camera.front" />
     <feature name="android.hardware.camera.flash" />
-    <feature name="android.hardware.location" />
-    <feature name="android.hardware.location.network" />
-    <feature name="android.hardware.sensor.accelerometer" />
-    <feature name="android.hardware.sensor.compass" />
-    <feature name="android.hardware.bluetooth" />
-    <feature name="android.hardware.touchscreen" />
-    <feature name="android.hardware.touchscreen.multitouch" />
-    <feature name="android.hardware.touchscreen.multitouch.distinct" />
-    <feature name="android.hardware.microphone" />
-    <feature name="android.hardware.screen.portrait" />
-    <feature name="android.hardware.screen.landscape" />
-    <feature name="android.software.app_widgets" />
-    <feature name="android.software.home_screen" />
-    <feature name="android.software.input_methods" />
-    <!-- Feature to specify if the device supports adding device admins. -->
-    <feature name="android.software.device_admin" />
-
-    <!-- This is added by FSL. -->
     <feature name="android.software.live_wallpaper" />
-
-
 </permissions>
-- 
1.8.0

